### PR TITLE
Add queue implementations for JS and Go

### PR DESCRIPTION
**Pull request by chidiwilliams**

### CODE REVIEW

File: queue/go/queue.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	return q.linkedList.RemoveFirst()
+}
+
+// Peek returns the element at the front of the queue.
+// It returns an error if the queue is empty.
+func (q Queue) Peek() (interface{}, error) {
+	return q.linkedList.Peek()
+}
+
+type doublyLinkedList interface {
```
Comment (oyekanmiayo) [inline queue/go/queue.go:26]: Since there's no implementation, will it be difficult to add driver code? @chidiwilliams

File: queue/js/queue.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  // Removes the element at the front of the queue
+  dequeue() {
+    return this.linkedList.removeFirst();
+  }
+
+  // Returns the element at the front of the queue
+  peek() {
+    return this.linkedList.peek();
+  }
+}
```
Comment (oyekanmiayo) [inline queue/js/queue.js:21]: Looks good. You should add driver code too @chidiwilliams

File: queue/go/queue.go (hunk 1 above)
Comment (chidiwilliams) [reply queue/go/queue.go:26]: Yeah, it will. That's why I couldn't add driver code here.

File: queue/js/queue.js (hunk 2 above)
Comment (chidiwilliams) [reply queue/js/queue.js:21]: I'll have to implement linked list on the file too to be able to add driver code.

Comment (oyekanmiayo) [reply queue/js/queue.js:21]: Can you implemented linkedlist differently and use it here? 

If yes, we can merge this for now and you can update when that's done.

Comment (chidiwilliams) [reply queue/js/queue.js:21]: Sure, could do that.

**Pull request merged**